src/OpenTelemetry.Exporter.OpenTelemetryProtocol.PersistentStorage/OtlpTraceExporterPersistentStorageExtensions.cs

```diff
@@ -0,0 +1,77 @@
+// <copyright file="OtlpTraceExporterPersistentStorageExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Exporter;
+using OpenTelemetry.Extensions.PersistentStorage.Abstractions;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Trace;
+
+/// <summary>
+/// Extension methods to simplify registering of the OpenTelemetry Protocol (OTLP) exporter
+/// with persistent storage.
+/// </summary>
+public static class OtlpTraceExporterPersistentStorageExtensions
+{
+    /// <summary>
+    /// Adds OpenTelemetry Protocol (OTLP) exporter to the TracerProvider
+    /// with access to persistent storage.
+    /// </summary>
+    /// <param name="builder"><see cref="TracerProviderBuilder"/> builder to use.</param>
+    /// <param name="name">Name which is used when retrieving options.</param>
+    /// <param name="configure">Callback action for configuring <see cref="OtlpExporterOptions"/>.</param>
+    /// <param name="persistentStorageFactory">Factory function to create a <see cref="PersistentBlobProvider"/>.</param>
+    /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddOtlpExporterWithPersistentStorage(
+        this TracerProviderBuilder builder,
+        string? name,
+        Action<OtlpExporterOptions> configure,
+        Func<IServiceProvider, PersistentBlobProvider> persistentStorageFactory)
+    {
+        Guard.ThrowIfNull(builder);
+        Guard.ThrowIfNull(persistentStorageFactory);
+
+        builder.ConfigureServices(services =>
+        {
+            // Try this if it doesn't work there are other ways to plug into options
+            services
+                  .AddOptions<OtlpExporterOptions>()
```

Comment:
@mic-max Shoot sorry you need to pass name here my bad...

`.AddOptions<OtlpExporterOptions>(name)`

You can remove the comment above too. Assuming you verified this works 😄

Reply:
![image](https://user-images.githubusercontent.com/15165028/219241220-47175d57-f655-4ac7-af69-4dff726e75b6.png)
oops, yes it is working still
![image](https://user-images.githubusercontent.com/15165028/219241274-3e5664a2-17a6-4ef6-bc8d-2e08bd154bed.png)